### PR TITLE
Fix reset method to call regenerateDeviceId.

### PR DESCRIPTION
**Pull request by f2prateek**
Previously we were calling clearUserProperties() which was incorrect.
Updated as per Amplitude's suggestion.

### CODE REVIEW

**Comment by f2prateek** [comment]
This was a a couple of lines of change so I decided to fix it.

**Comment by ladanazita** [comment]
LGTM 👍

File: src/test/java/com/segment/analytics/android/integrations/amplitude/AmplitudeTest.java [hunk 1]
```diff
@@ -246,7 +247,10 @@
   @Test public void reset() {
     integration.reset();
 
-    verify(amplitude).clearUserProperties();
+    verify(amplitude).regenerateDeviceId();
+
+    // Previously we called clearUserProperties() which was incorrect.
+    verify(amplitude, never()).clearUserProperties();
```
Comment (ladanazita) [inline src/test/java/com/segment/analytics/android/integrations/amplitude/AmplitudeTest.java]: Curious - why do we have to verify this is never called?

Comment (f2prateek) [inline src/test/java/com/segment/analytics/android/integrations/amplitude/AmplitudeTest.java]: Not strictly needed, but mostly to ensure we aren't still calling it. If i didn't have this, my test would still pass if I hadn't removed `clearUserProperties` from the integration. Also prevents a regression in the future.

**Pull request merged**